glossary/build-trigger.md

```diff
@@ -0,0 +1,9 @@
+---
```

Comment:
This is a good start, I'd like to make a guide doc for build triggers in general soon that this can link to.